demos/dotnet/dotnet/NugetAction.cs

```diff
@@ -0,0 +1,188 @@
+            {
+                Console.WriteLine("Failed to get nuget or restore packages.");
+                return false;
+            }
+            return true;
+        }
+
+        public static bool DownloadNugetAction(ProjectProperties properties, string nugetFile)
+        {
+            CreateDefaultProjectJson(properties);
+            CreateNugetConfig(properties);
+
+            var client = new HttpClient();
+            var requestUri = new Uri(@"http://dist.nuget.org/win-x86-commandline/v3.1.0-beta/nuget.exe",
+                UriKind.Absolute);
+
+            var sourceStreamTask = client.GetStreamAsync(requestUri);
+            try
+            {
+                sourceStreamTask.Wait();
+            }
+            catch (AggregateException exception)
+            {
+                foreach (var ex in exception.InnerExceptions)
+                {
+                    Console.WriteLine(ex.Message);
+                    return false;
+                }
+            }
+
+            using (var sourceStream = sourceStreamTask.Result)
+            using (var destinationStream = new FileStream(nugetFile, FileMode.Create, FileAccess.Write))
+            {
+                var buffer = new byte[1024];
+                while (true)
+                {
+                    var read = sourceStream.Read(buffer, 0, buffer.Length);
+                    if (read < 1) break;
+                    destinationStream.Write(buffer, 0, read);
+                }
+            }
+
+            return true;
+        }
+
+        private static void CreateDefaultProjectJson(ProjectProperties properties)
+        {
+            var fileName = Path.Combine(properties.ToolsDirectory, "project.json");
+            var fs = new FileStream(fileName, FileMode.Create);
+            using (var file = new StreamWriter(fs, Encoding.UTF8))
+            {
+                file.WriteLine(@"{");
+                file.WriteLine(@"    ""dependencies"": {");
+
+                for (var index = 0; index < properties.Packages.Count; index++)
+                {
+                    var package = properties.Packages[index];
+                    file.Write(@"        ");
+                    file.Write(package);
+                    if (index < properties.Packages.Count - 1)
+                    {
+                        file.WriteLine(",");
+                    }
+                    else
+                    {
+                        file.WriteLine();
+                    }
+                }
+                file.WriteLine(@"    },");
+                file.WriteLine(@"    ""frameworks"": {");
+                file.WriteLine(@"        ""dnxcore50"": { }");
+                file.WriteLine(@"    }");
+                file.WriteLine(@"}");
+
+                //"runtimes": {
+                //"win7-x86": { },
+                //"win7-x64": { }
+                //},
+            }
+        }
+
+        private static void CreateNugetConfig(ProjectProperties properties)
```

Comment:
If this were nuget.exe, is it responsible for Create Nuget Config? Or is that someone else's job?